File: 1-two-sum/two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer,Integer> map = new HashMap<>();
        for(int i=0;i<nums.length;i++){
            int bal = target-nums[i];
            if(map.containsKey(bal)){
                return new int[]{map.get(bal),i};
            }
            map.put(nums[i],i);
        }
        return new int[]{-1,-1};
    }
}

File: 1-two-sum/two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        prev={}

        for i in range(len(nums)):
            n=nums[i]
            dif = target - n
            if dif in prev:
                return [prev[dif],i]
            prev[n]=i
        
        return [-1,-1]